crates/sqlbuiltins/src/functions/scalars/hashing.rs

```diff
@@ -0,0 +1,100 @@
+use std::hash::{Hash, Hasher};
+
+use fnv::FnvHasher;
+use siphasher::sip::SipHasher24;
+
+use super::*;
+
+pub struct SipHash;
+
+impl ConstBuiltinFunction for SipHash {
+    const NAME: &'static str = "siphash";
+    const DESCRIPTION: &'static str =
+        "Calculates a 64bit non-cryptographic hash (SipHash24) of the value.";
```

Comment:
is this row or columnar hashing? 

from the implementation, it looks like it's vectorized (columnar), but i'm not sure if that is clear from the slts or the description here.

Reply:
I think it's row based (and it wouldn't make much sense if it weren't for the partitioning case,) and is consistent with the other hashing/checksum functions. My intention was that that the word "value" (rather than column or values) would indicate this, but if there's a better term of art, I'm not wed to this wording.